[IE Samples] Update docs for C++ samples (#9937)
openvinotoolkit/openvino: samples/cpp/classification_sample_async/README.md

# Image Classification Async C++ Sample {#openvino_inference_engine_samples_classification_sample_async_README}

> This sample demonstrates how to do inference of image classification networks using Asynchronous Inference Request API.  

Reviewer: ```suggestion
This sample demonstrates how to do inference of image classification models using Asynchronous Inference Request API.  
```

# Image Classification Async C++ Sample {#openvino_inference_engine_samples_classification_sample_async_README}

This sample demonstrates how to do inference of image classification models using Asynchronous Inference Request API.  
Models with only one input and output are supported.

In addition to regular images, the sample also supports single-channel `ubyte` images as an input for LeNet model.

> The following Inference Engine C++ API is used in the application:

Reviewer 2: ```suggestion
The following C++ API is used in the application:
```

| Feature | API | Description |
| :--- | :--- | :--- |

> | Asynchronous Infer | `ov::InferRequest::start_async`, `ov::InferRequest::set_callback` | Do asynchronous inference with callback |

Author: We can have something like this `Manage the model, operate with its batch size. Set batch size using input image count.` in description column. Maybe we should use a semicolon instead of a comma?

| Model Operations | `ov::Output::get_shape`, `ov::set_batch` |  Manage the model, operate with its batch size. Set batch size using input image count. |
| Infer Request Operations | `ov::InferRequest::get_input_tensor` | Get an input tensor. |
| Tensor Operations | `ov::shape_size`, `ov::Tensor::data` | Get a tensor shape size and its data. |

Basic OpenVINO™ Runtime API is covered by [Hello Classification C++ sample](../hello_classification/README.md).

| Options | Values |
| :--- | :--- |
| Validated Models | [alexnet](@ref omz_models_model_alexnet), [googlenet-v1](@ref omz_models_model_googlenet_v1) |
| Model Format | OpenVINO™ toolkit Intermediate Representation (\*.xml + \*.bin), ONNX (\*.onnx) |
| Supported devices | [All](../../../docs/OV_Runtime_UG/supported_plugins/Supported_Devices.md) |
| Other language realization | [Python](../../../samples/python/classification_sample_async/README.md) |

## How It Works

At startup, the sample application reads command line parameters and loads the specified model and input images (or a

> folder with images) to the Inference Engine plugin. The batch size of the model is set according to the number of read images. The batch mode is an independent attribute on the asynchronous mode. Asynchronous mode works efficiently with any batch size.

Reviewer 2: ```suggestion
folder with images) to the OpenVINO™ Runtime plugin. The batch size of the model is set according to the number of read images. The batch mode is an independent attribute on the asynchronous mode. Asynchronous mode works efficiently with any batch size.
```
